include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
+#define GSL_USE_STD_BYTE 1
+
+#else
+
+#define GSL_USE_STD_BYTE 0
+
+#endif // _HAS_STD_BYTE
+
 #endif           // _MSC_VER
```

Comment:
I would put a:
```c++
#else
#define GSL_USE_STD_BYTE 0
```
here, to avoid clang users getting `-Wundef` warnings when we test `GSL_USE_STD_BYTE` on line 55.